docs/paper/dev/api/components.md

```diff
@@ -0,0 +1,191 @@
+   to make code less verbose */
+final Component component = text()
+    .content("Hello").color(color(0x13f832))
+    .append(text(" world!", GREEN))
+    .build();
+```
+
+:::info In-Depth Documentation
+
+For complete documentation on the Adventure Component API Paper and Velocity use, please look at the
+[Adventure documentation](https://docs.advntr.dev).
+
+:::
+
+## MiniMessage
+
+Paper and Velocity include the MiniMessage library which is a string representation of components. If you prefer working with
+strings rather than objects, MiniMessage is vastly superior to the legacy string format. It can utilize the tree
+structure for style inheritance and can represent the more complex component types while legacy cannot.
+
+```java
+final Component component = MiniMessage.miniMessage().deserialize(
+    "<#438df2><b>This is the parent component; its style is " +
+    "applied to all children.\n<u><!b>This is the first child, " +
+    "which is rendered after the parent</!b></u><key:key.inventory></b></#438df2>"
+);
+```
+
+We recommend using this format for user-facing input such as commands or configuration values.
+
+:::info In-Depth Documentation
+
+MiniMessage is a part of Adventure, and you can find its documentation on [Adventure's documentation](https://docs.advntr.dev/minimessage/index.html).
+
+:::
+
+:::tip
+
+MiniMessage as a [WebViewer](https://webui.advntr.dev/) which is useful for constructing more complicated components and seeing the results in real time
+
+:::
+
+## JSON Format
+
+Components can be serialized and deserialized from a standard JSON format. This format is used
+in vanilla in various commands which accept component arguments like `/tellraw`. Below is a simple example
+of this format.
+
+```json
+{
+  "text": "This is the parent component; its style is applied to all children.\n",
+  "color": "#438df2",
+  "bold": true,
+  "extra": [
+    {
+      "text": "This is this first child, which is rendered after the parent",
+      "underlined": true,
+      // This overrides the parent's "bold" value just for this component
+      "bold": false
+    },
+    {
+      // This is a keybind component which will display the client's keybind for that action
+      "keybind": "key.inventory"
+    }
+  ]
+}
+```
+
+:::info In-Depth Documentation
+
+The JSON format is fully documented on the [Minecraft Wiki](https://minecraft.fandom.com/wiki/Raw_JSON_text_format). 
+
+:::
+
+:::tip
+
+There are online tools to make generating this format much easier like [JSON Text Generator](https://minecraft.tools/en/json_text.php).
+
+:::
+
+## Serializers
+
+Paper and Velocity come bundled with different serializers for converting between Components and other forms
+of serialized text.
+
+### [GsonComponentSerializer](https://jd.advntr.dev/text-serializer-gson/latest)
+
+Converts between `Component` and Gson's `JsonElement`. This conversion is lossless and is the preferred form of serialization
```

Comment:
`String` is technically the serialized form that `GsonComponentSerializer` produces - there's just convenience methods that use `JsonElement`.